File: sic-app/src/app/core/services/work-package.service.ts
// src/app/core/services/work-package.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import type { WorkPackageResponse, WorkPackageRequest } from '../model/phase.model';

@Injectable({
  providedIn: 'root',
})
export class WorkPackageService {
  private http = inject(HttpClient);
  private apiBaseUrl = environment.apiBaseUrl;

  getWorkPackageById(id: string): Observable<WorkPackageResponse> {
    return this.http.get<WorkPackageResponse>(`${this.apiBaseUrl}/api/pm/work-packages/${id}`);
  }

  createWorkPackage(data: WorkPackageRequest): Observable<WorkPackageResponse> {
    return this.http.post<WorkPackageResponse>(`${this.apiBaseUrl}/api/pm/work-packages`, data);
  }

  updateWorkPackage(id: string, data: WorkPackageRequest): Observable<WorkPackageResponse> {
    return this.http.put<WorkPackageResponse>(`${this.apiBaseUrl}/api/pm/work-packages/${id}`, data);
  }

  deleteWorkPackage(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiBaseUrl}/api/pm/work-packages/${id}`);
  }

  // ✅ 新增: โหลด Work Packages ตาม Milestone ID
  getWorkPackagesByMilestoneId(milestoneId: string): Observable<WorkPackageResponse[]> {
    return this.http.get<WorkPackageResponse[]>(
      `${this.apiBaseUrl}/api/pm/work-packages/milestone/${milestoneId}`
    );
  }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/service/WorkPackageService.java
package com.softinter.sicapi.service;


import java.util.List;
import java.util.UUID;

import com.softinter.sicapi.dto.request.WorkPackageRequest;
import com.softinter.sicapi.dto.response.WorkPackageResponse;

public interface WorkPackageService {
    List<WorkPackageResponse> getWorkPackagesByMilestoneId(UUID milestoneId);
    WorkPackageResponse getWorkPackageById(UUID wpId);
    WorkPackageResponse createWorkPackage(WorkPackageRequest request);
    WorkPackageResponse updateWorkPackage(UUID wpId, WorkPackageRequest request);
    void deleteWorkPackage(UUID wpId);
}